framework/core/transform_error.ts

```diff
@@ -37,14 +43,41 @@ const errorTemplate = (message: string) => `
 
 events.on("transform", (e) => {
   if (e.status === "failure") {
-    const err = e.error as { message: string; position: [line: number, column: number] };
-    modal.innerHTML = errorTemplate(err.message);
+    console.log(e);
+    const err = e.error as { message: string; stack: string; location: Array<number> };
+    const code = formatCode(err.message, e.sourceCode as string, err.location[1], err.location[0]);
+    const stack = err.stack.split("\n").map((v) => v.trim());
+    const stackStr = stack.filter((v) => !v.includes("wasm://wasm")).reduce((res, cur) => res + "\n" + cur);
+    modal.innerHTML = errorTemplate(err.message, code, stackStr);
     modal.style.display = "block";
   } else {
     modal.style.display = "none";
   }
 });
 
+function formatCode(message: string, sourceCode: string, column: number, line: number): string {
+  if (message === "unreachable") {
+    return message;
+  }
+  let sourceCodeArr = sourceCode.split("\r\n").map((val, index) => {
+    return 1 + index + " | " + val;
+  });
+  const indexLen = line.toString().length;
+  const mark = getBlank(indexLen * 2) + " | " + getBlank(column) + "^";
+  sourceCodeArr = sourceCodeArr.slice(line - 3, line + 2);
+  sourceCodeArr.splice(3, 0, mark);
+  const formatStr = sourceCodeArr.reduce((res, cur) => res + "\r\n" + cur);
+  return formatStr;
+}
+
+function getBlank(num: number) {
```

Comment:
you can use `" ".repeat(num)` instead, since the message is in the `pre` tag that will keep the spaces

Reply:
great!